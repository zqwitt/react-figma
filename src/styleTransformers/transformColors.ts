import { Color } from '../types';
import * as parse from 'color-parse';

export const colorToRGB = (color: Color): RGB => {
    const parsed = parse(color);
    if (parsed && parsed.space === 'rgb') {
        return {
            r: parsed.values[0] / 255.0,
            g: parsed.values[1] / 255.0,
            b: parsed.values[2] / 255.0
        };
    }
};

export const colorToRGBA = (color: Color, opacity: number): RGBA => {
    return {
        ...colorToRGB(color),
        a: opacity
    };
};
