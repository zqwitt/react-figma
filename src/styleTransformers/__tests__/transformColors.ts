import { colorToRGB, colorToRGBA } from '../transformColors';

describe('transformColors', () => {
    it('colorToRGB #9b9fff', () => {
        const result = colorToRGB('#9b9fff');
        expect(result).toMatchSnapshot();
    });

    it('colorToRGB #ff0000', () => {
        const result = colorToRGB('#ff0000');
        expect(result).toMatchSnapshot();
    });

    it('colorToRGBA #ff0000 0.5', () => {
        const result = colorToRGBA('#ff0000', 0.5);
        expect(result).toMatchSnapshot();
    });

    it('colorToRGB red', () => {
        const result = colorToRGB('red');
        expect(result).toMatchSnapshot();
    });

    it('colorToRGB aliceblue', () => {
        const result = colorToRGB('aliceblue');
        expect(result).toMatchSnapshot();
    });

    it('colorToRGB rgb(255, 0, 255)', () => {
        const result = colorToRGB('rgb(255, 0, 255)');
        expect(result).toMatchSnapshot();
    });

    it('colorToRGB rgba(255, 255, 255, 1.0)', () => {
        const result = colorToRGB('rgba(255, 255, 255, 1.0)');
        expect(result).toMatchSnapshot();
    });

    it('colorToRGB #ff00ff00', () => {
        const result = colorToRGB('#ff00ff00');
        expect(result).toMatchSnapshot();
    });

    it('colorToRGB transparent', () => {
        const result = colorToRGB('transparent');
        expect(result).toMatchSnapshot();
    });
});
