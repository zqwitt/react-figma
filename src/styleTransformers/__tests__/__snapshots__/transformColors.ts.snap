// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transformColors colorToRGB #9b9fff 1`] = `
Object {
  "b": 1,
  "g": 0.6235294117647059,
  "r": 0.6078431372549019,
}
`;

exports[`transformColors colorToRGB #ff0000 1`] = `
Object {
  "b": 0,
  "g": 0,
  "r": 1,
}
`;

exports[`transformColors colorToRGB #ff00ff00 1`] = `
Object {
  "b": 1,
  "g": 0,
  "r": 1,
}
`;

exports[`transformColors colorToRGB aliceblue 1`] = `
Object {
  "b": 1,
  "g": 0.9725490196078431,
  "r": 0.9411764705882353,
}
`;

exports[`transformColors colorToRGB red 1`] = `
Object {
  "b": 0,
  "g": 0,
  "r": 1,
}
`;

exports[`transformColors colorToRGB rgb(255, 0, 255) 1`] = `
Object {
  "b": 1,
  "g": 0,
  "r": 1,
}
`;

exports[`transformColors colorToRGB rgba(255, 255, 255, 1.0) 1`] = `
Object {
  "b": 1,
  "g": 1,
  "r": 1,
}
`;

exports[`transformColors colorToRGB transparent 1`] = `
Object {
  "b": 0,
  "g": 0,
  "r": 0,
}
`;

exports[`transformColors colorToRGBA #ff0000 0.5 1`] = `
Object {
  "a": 0.5,
  "b": 0,
  "g": 0,
  "r": 1,
}
`;
